components/pkg-export-tar/src/build.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs as stdfs;
+#[cfg(target_os = "linux")]
+use std::os::unix::fs::symlink;
+use clap;
+use common;
+use common::command::package::install::InstallSource;
+use common::ui::{UI, Status};
+use tempdir::TempDir;
+use std::path::Path;
+use hcore::package::PackageIdent;
+use error::Result;
+use hcore::PROGRAM_NAME;
+use hcore::fs::{CACHE_ARTIFACT_PATH, CACHE_KEY_PATH, cache_artifact_path, cache_key_path};
+
+use super::{VERSION, BUSYBOX_IDENT};
+
+use rootfs;
+
+const DEFAULT_HAB_IDENT: &'static str = "core/hab";
+const DEFAULT_LAUNCHER_IDENT: &'static str = "core/hab-launcher";
+const DEFAULT_SUP_IDENT: &'static str = "core/hab-sup";
+
+/// The specification for creating a temporary file system build root, based on Habitat packages.
+///
+/// When a `BuildSpec` is created, a `BuildRoot` is returned which can be used to produce exported
+/// images, archives, etc.
+#[derive(Debug)]
+pub struct BuildSpec<'a> {
+    /// A string representation of a Habitat Package Identifer for the Habitat CLI package.
+    pub hab: &'a str,
+    /// A string representation of a Habitat Package Identifer for the Habitat Launcher package.
+    pub hab_launcher: &'a str,
+    /// A string representation of a Habitat Package Identifer for the Habitat Supervisor package.
+    pub hab_sup: &'a str,
+    /// The Builder URL which is used to install all service and extra Habitat packages.
+    pub url: &'a str,
+    /// The Habitat release channel which is used to install all service and extra Habitat
+    /// packages.
+    pub channel: &'a str,
+    /// The Builder URL which is used to install all base Habitat packages.
+    pub base_pkgs_url: &'a str,
+    /// The Habitat release channel which is used to install all base Habitat packages.
+    pub base_pkgs_channel: &'a str,
+    /// A list of either Habitat Package Identifiers or local paths to Habitat Artifact files which
+    /// will be installed.
+    pub idents_or_archives: Vec<&'a str>,
```

Comment:
If multiple artifacts are not supported, we should make the name of this field singular, and remove the `Vec` from the type, just to make it super clear.

Reply:
Fixed!